### ISSUE TITLE
Access local Zotero through JSON-RPC

### ISSUE THREAD

**Issue by retorquere** [LaTeXing/LaTeXing#153]
I maintain the Zotero Better BibTeX plugin. I could add a local JSON-RPC endpoint as I have done for Atom and Brackets so that LaTeXing could access the locally running Zotero (including the enhanced citation key management). This would mean syncing wouldn't be necessary anymore to get newly added references. Interesting?

**Comment by sdaza** [comment]
Yes!

**Comment by retorquere** [comment]
Just tell me what you need. I would strongly prefer to stick to json-rpc -
it's simple, and I have the plumbing ready. I can whip up any bunch of
methods you need, they'll be namespaced to latexing, so go wild.

**Comment by sdaza** [comment]
I am not a latexing maintainer, only a user! When using zotero and latexing, I don't get the institutional citations right. I guess that using Better BibTeX will solve the problem :)

**Comment by csch0** [comment]
thanks @sdaza. I can see the advantage of using Better BibTeX, but also the disadvantage for LaTeXing requires now Zotero installed on the machine. Maintaining multiple interfaces for one service isn't recommended in my opinion.

Will have a look for Better BibTeX asap and come back to you @retorquere

**Comment by retorquere** [comment]
Ah, I see. Well, BBT auto export ought to achieve most of what you want.
For larger libraries, a live connection between latexing and BBT is
preferable for performance, but since 0.11 is very good anyhow.

**Comment by retorquere** [comment]
The upside of a connection through bbt over connecting to the zotero server
is that:

* it's much, much faster (real-time lookups would be a cinch, even with
very large libraries)
* it's available offline
* the citation key generator is much better, and safer.

I can mimic the zotero api if need be. Doesn't have my preference, but it
can be done.

**Comment by sdaza** [comment]
When I use the BBT auto export feature + global_bib_file_path for all my references, I am not able to generate specific ref.bib files for my different projects. This is problematic because I share my work with others, and I don't want to share a huge .bib file.  

```  
"global_bib_file_path": ["/Users/sdaza/Dropbox/sdaza.bib"],
"bibname": "ref.bib"
```

**Comment by sdaza** [comment]
It is like importing references doesn't work when you use a global bib file.

**Comment by csch0** [comment]
ok @sdaza, fixed one issue related to this in the upcoming 1.2.13, not sure I can complete the release process today but for sure tomorrow. Lets create a new issue if it still exists in that version.

**Comment by retorquere** [comment]
@sdaza we can discuss BBT elsewhere, but you can in fact set up multiple exports each exporting different parts of your library to separate files using BBT.

**Comment by sdaza** [comment]
Yes, I know, but that means creating different collections/tags in zotero. I am just want to create my references as I write using Latexing.

**Comment by retorquere** [comment]
For that I'll need help from the LaTeXing team. Let's give @csch0 some time to review the options.

**Comment by sdaza** [comment]
@csch0 Just to know. Do you have a tentative date for releasing the 1.2.13 version?
